encode.go

```diff
@@ -28,8 +28,6 @@ const (
 type Encoder struct {
 	writer                     io.Writer
 	opts                       []EncodeOption
-	indent                     int
```

Comment:
The Encoder previously had two fields, `indent` and `indentNum`, but `indentNum` was not being used anywhere. Since the name `indentNum` is used elsewhere, I migrated the functionality of the `indent` field to the `indentNum` field and removed the `indent` field.